src/gateway.h

```diff
@@ -29,6 +29,7 @@ struct gateway
 	struct registry *registry;   /* Register of existing databases */
 	struct raft *raft;           /* Raft instance */
 	struct leader *leader;       /* Leader connection to the database */
+	bool leadership_lost;        /* Flag if node lost leadership */
```

Comment:
Looking closer, it seems that the ```gateway``` object doesn't really make use of this field on its own, while it's the ```conn``` object that uses it. I'd suggest to move the field to the ```conn``` object instead.

Reply:
okay